scripts/e2e-test.sh

```
@@ -1,507 +1,219 @@
+  python3 "$HELPER" register 2>&1 | sed 's/^/  /' || info "User may already exist"
+  sleep 1
+  python3 "$HELPER" api-key > "$E2E_DATA" 2>&1 || { fail "Failed to get credentials"; cat "$E2E_DATA"; exit 1; }
+  CLIENT_ID=$(python3 -c "import json; print(json.load(open('$E2E_DATA'))['clientId'])")
+  CLIENT_SECRET=$(python3 -c "import json; print(json.load(open('$E2E_DATA'))['clientSecret'])")
+  ok "Client credentials: $CLIENT_ID"
+
+  step "Creating 3 test items"
+  ITEMS_JSON=$(python3 "$HELPER" create-items 3 2>&1) || true
+  echo "$ITEMS_JSON" | head -3 | sed 's/^/  /'
+  ITEM_A_ID=$(echo "$ITEMS_JSON" | python3 -c "import json,sys; d=json.load(sys.stdin); print(d.get('items',[{}])[0].get('id',''))" 2>/dev/null || echo "")
+  [ -n "$ITEM_A_ID" ] && info "Item A ID for rename test: ${ITEM_A_ID::16}..." || info "Item ID not captured"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)